.gitignore

```diff
@@ -24,3 +24,5 @@ eggs/
 *.egg
 .DS_Store
 .vscode/*
+*.so
+*-checkpoint.ipynb
```

Comment:
Are `.ipynb` files generated by any of the scripts added here? If not please remove it.

Reply:
Thanks, I'll remove this change.

The `*-checkpoint.ipynb` files are generated by Jupyter Notebook servers automatically (similar to how vim generates .swp files -- but doesn't delete them after Jupyter Notebook is shut down). But these Jupyter Notebooks were added primarily during our internal analyses and I did not add them here.